metrics.go

```diff
@@ -23,20 +29,37 @@ type metrics struct {
 	dealsTotal           asyncint64.Gauge
 	dealsCoverage        asyncfloat64.Gauge
 
-	snapshotLock   sync.RWMutex
-	countsByStatus map[Status]int64
+	countsByStatusLock sync.RWMutex
+	countsByStatus     map[Status]int64
 
+	// totalParticipantCount is the total number of state market participants retrieved from
+	// FileCoin API stored as int64.
 	totalParticipantCount atomic.Value
-	totalDealCount        atomic.Value
-
-	reachableWithNonZeroDeals      int
-	knownByIndexer                 int
-	dealCountDiscoverableByIndexer int64
+	// totalDealCount is the total number of state market deals retrieved from FileCoin API  stored
+	// as int64.
+	totalDealCount atomic.Value
+	// totalAddressableParticipantsWithNonZeroDeals is the total number of state market participants
+	// which: 1) had non-nil peer.AddrInfo on FileCoin API, and 2) have made at least one deal
+	// stored as int64.
+	totalAddressableParticipantsWithNonZeroDeals atomic.Value
+	// totalAddressableParticipants s the total number of state market participants which had
+	// non-nil peer.AddrInfo on FileCoin API.
+	totalAddressableParticipants atomic.Value
+	// totalParticipantsKnownByIndexer is the total number of state market participants that are
+	// listed as a provider by network indexer stored as int64.
+	totalParticipantsKnownByIndexer atomic.Value
+	// totalDealCountByParticipantsKnownToIndexer is the total number of deals made by the state
+	// market participants that are listed as a provider by network indexer stored as int64.
+	totalDealCountByParticipantsKnownToIndexer atomic.Value
 }
 
 func (m *metrics) start() error {
 	m.totalParticipantCount.Store(int64(0))
 	m.totalDealCount.Store(int64(0))
+	m.totalAddressableParticipantsWithNonZeroDeals.Store(int64(0))
+	m.totalAddressableParticipants.Store(int64(0))
+	m.totalParticipantsKnownByIndexer.Store(int64(0))
+	m.totalDealCountByParticipantsKnownToIndexer.Store(int64(0))
```

Comment:
These additions are great, exactly the questions I was looking to answer. :tada: